src/middlewared/middlewared/schema.py

```diff
@@ -437,9 +437,9 @@ def clean(self, value):
                     try:
                         value[index] = i.clean(v)
                         found = True
+                        break
```

Comment:
What are you trying to achieve here? Allow different types in the list?
What if none of them match?

Reply:
Then we raise an exception. I initially did this because i was thinking of handling extensions differently which included passing a list of dictionaries which obviously failed validation because right now we only support one type of item in list schema. I kept this around if you think we should allow this behaviour. I can have it removed and if none of them match, we raise an exception

Reply:
Ok, understood. No I think that is fine. We should probably add these cases to the unittests though, as we are changing the behavior